transit.md

```diff
@@ -176,13 +176,48 @@ provided by the Transit instance include the locally-configured relay, along
 with the dynamically-determined direct hints. Both should be delivered to the
 peer.
 
-## Canonical hint encodings
+
+## Canonical abilities encodings
 
 The transit protocol relies on an existing secured (but possibly low-bandwidth)
 communication channel to exchange the abilities and hints. Thus, it has no
 influence over how they are encoded. However, we make a suggestion using JSON
 messages for other protocols to use.
 
+Abilities are encoded as a list, each item having a `type` tag. An entry of type
+`relay-v1` may additionally contain a `url_hints` boolean attribute. Ability may
+appear more than once in the list.
+
+```json
+[
+  {
+    "type": "<string, one of {direct-tcp-v1, relay-v1, tor-tcp-v1}>"
+  }
+]
+```
+
+Example for the full abilities set:
+
+```json
+[
+  {
+    "type": "direct-tcp-v1"
+  },
+  {
+    "type": "relay-v1",
+    "url_hints": true
+  },
```

Comment:
Now I'm really confused: what is this communicating (i.e. why aren't the actual URLs here)?

Reply:
Because this is only the *ability*, telling the peer that you are *capable of* sending hints with the new format. Because if you know your peer doesn't, you then can save the effort.

But I'm not entirely sure where the confusion comes from, since this is an existing concept of the protocol that I didn't really change. Maybe it's because the current transfer protocol sends both abilities and hints in a single message (IIRC), which may reduce one round trip but kind of limits the usefulness of a prior capability exchange.